app/models/spree_avatax/sales_invoice.rb

```diff
@@ -25,6 +25,11 @@ def generate(order)
 
       return if order.completed? || !SpreeAvatax::Shared.taxable_order?(order)
 
+      taxable_records = order.line_items + order.shipments
+      taxable_records.each do |taxable_record|
+        taxable_record.update_column(:pre_tax_amount, taxable_record.discounted_amount.round(2))
+      end
```

Comment:
That seems like a reasonable concern and, all things considered, I think that falls under the "bad name" category because I think the most pragmatic way to make that better would be to rename this method `calculate_and_set_taxes` (or something like that) and deprecate the name `generate`.

But in any case this PR isn't a significant change as far as how it mutates the order so how about we open an issue or a separate PR around that.